File: React:TyeScript- FE/src/api/models/film.ts
import { Genre } from "./genre";
import { Review } from "./review";

export interface FilmBase {
  id: string;
  title: string;
}

export interface Film extends FilmBase {
  summary: string;
  shortSummary: string;
  runtime: number;
  releaseDate: string;
  distributor: string;
  genres: Genre[];
  posterUrl: string;
  reviews: Review[];
  overallScore : number;
}

File: C#-api/DataLayer/Models/Film.cs
namespace films_assignment_api.DataLayer.Models
{
    public class Film : BaseModel
    {
        public string Title { get; set; }
        public string Summary { get; set; }
        public string ShortSummary { get; set; }
        public int Runtime { get; set; }
        public string ReleaseDate { get; set; }
        public string Distributor { get; set; }
        public List<string> Genres { get; set; }
    }
}
